src/bullet_utils/wrapper.py

```diff
@@ -263,6 +264,28 @@ def get_state(self):
 
         return q, dq
 
+    def get_imu_frame_position_velocity(self):
+        """Returns the position and velocity of IMU frame. Note that the velocity is expressed in the IMU frame.
+
+        Returns:
+            np.array((3,1)): IMU frame position expressed in world.
+            np.array((3,1)): IMU frame velocity expressed in IMU frame.
+        """ 
+        q, dq = self.get_state()
+        
+        base_inertia_pos, base_inertia_quat = pybullet.getBasePositionAndOrientation(self.robot_id)
+        # Get transform between inertial frame and link frame in base
+        base_stat = pybullet.getDynamicsInfo(self.robot_id, -1)
+        base_inertia_link_pos, base_inertia_link_quat = pybullet.invertTransform(base_stat[3], base_stat[4])
+        base_pos, base_quat = pybullet.multiplyTransforms(base_inertia_pos, base_inertia_quat, base_inertia_link_pos, base_inertia_link_quat)
+        rot_base_to_world = np.array(pybullet.getMatrixFromQuaternion(base_quat)).reshape((3, 3))
+
+        base_angvel = dq[3:6]
```

Comment:
Is this angular velocity in base or world frame? Seems to be in base from how you use it to compute the imu_velocity, but then the imu_position seems to use q[:3] as if that position is in world frame.